src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -920,76 +828,69 @@ else if (ctx.plus.size() > 0)
     }
 
     @Override
-    public Void visitSingleType(JsoniqParser.SingleTypeContext ctx) {
+    public Node visitSingleType(JsoniqParser.SingleTypeContext ctx) {
         if (ctx.item == null)
-            this.currentExpression = new FlworVarSingleType(createMetadataFromContext(ctx));
-        else {
-            AtomicTypes item = FlworVarSingleType.getAtomicType(ctx.item.getText());
-            if (ctx.question.size() > 0)
-                this.currentExpression = new FlworVarSingleType(item, true, createMetadataFromContext(ctx));
-            else
-                this.currentExpression = new FlworVarSingleType(item, createMetadataFromContext(ctx));
-        }
-        return null;
+            return new FlworVarSingleType(createMetadataFromContext(ctx));
+
+        AtomicTypes item = FlworVarSingleType.getAtomicType(ctx.item.getText());
+        if (ctx.question.size() > 0)
+            return new FlworVarSingleType(item, true, createMetadataFromContext(ctx));
+        else
```

Comment:
else can be removed

Reply:
Yes, adding {} automatically will be best once it is merge. Thanks!